### PR TITLE
Update document for zeros_like, ones_like, softmax_cross_entropy, c_array, c_str

### PR DESCRIPTION
**Pull request by yuruofeifei**
Update document for zeros_like, ones_like, softmax_cross_entropy, c_array, c_str and fix some docstring

### CODE REVIEW

File: src/operator/tensor/init_op.cc [hunk 1]
```diff
@@ -56,7 +70,21 @@ NNVM_REGISTER_OP(zeros_like)
 .add_argument("data", "NDArray-or-Symbol", "The input");
 
 NNVM_REGISTER_OP(ones_like)
-.MXNET_DESCRIBE("Return an array of ones with the same shape and type as the input array.")
+.describe(R"code(Return an array of ones with the same shape and type
+as the input array.
+
+Examples::
+
+  >>> x = mx.nd.zeros((2,3))
```
Comment (piiswrong) [inline src/operator/tensor/init_op.cc]: docs in backend are shared for all frontends, so shouldn't have python grammar. See other APIs for example http://mxnet.io/api/python/ndarray.html#mxnet.ndarray.ROIPooling

**Comment by nswamy** [comment]
@zackchase @madjam please review

File: src/operator/loss_binary_op.cc [hunk 2]
```diff
@@ -9,7 +9,25 @@ namespace mxnet {
...
+   cross_entropy(\mathbf{z}) = \sum_{c=1}^C log(softmax((\mathbf{z})_c) * target_c
+
+This function will convert labels to one_hot encoding first, so
+
+.. math::
+   texttt{cross\_entropy}(\mathbf{z}) = log(softmax(\mathbf{z})_c)
+
+Example::
+
+  >>> data = mx.nd.array([[1, 2, 3], [11, 7, 5]])
```
Comment (nswamy) [inline src/operator/loss_binary_op.cc]: Please don't have python specific constructs in the example here. The examples in the C++ documentation must be language neutral, they will show up in other language bindings as well.

**Comment by nswamy** [comment]
Also please run make lint as described in [How to make a pull request for docs](https://s3.amazonaws.com/mxnetdoc/API+Doc+Contributions%281%29.pdf)

File: python/mxnet/base.py [hunk 3]
```diff
@@ -120,7 +132,7 @@ def c_array(ctype, values):
 
     Parameters
     ----------
-    ctype : ctypes data type
+    ctype : ctypes data type such as mx_float
```
Comment (zackchase) [inline python/mxnet/base.py]: Is this the right use of the formal parameter type specification - @piiswrong @mli can you comment?

File: src/operator/loss_binary_op.cc [hunk 4]
```diff
@@ -9,7 +9,25 @@ namespace mxnet {
...
 
 NNVM_REGISTER_OP(softmax_cross_entropy)
-.MXNET_DESCRIBE("Calculate cross_entropy(data, one_hot(label))")
+.describe(R"code(Calculate cross entropy loss
+between the softmax output of input arrays and one_hot encoding of labels.
+
+.. math::
+   cross_entropy(\mathbf{z}) = \sum_{c=1}^C log(softmax((\mathbf{z})_c) * target_c
+
+This function will convert labels to one_hot encoding first, so
```
Comment (zackchase) [inline src/operator/loss_binary_op.cc]: If you mean "one-hot" in English, use a hyphen. If you mean to refer to the code (to a function) then put in backticks.

**Comment by jiajiechen** [comment]
There might be merge conflict with https://github.com/dmlc/mxnet/pull/5945 because that one also has softmax_cross_entropy..

File: python/mxnet/base.py (hunk 3 above)
Comment (piiswrong) [inline python/mxnet/base.py]: no.
such as mx_float should go into comment

**Comment by nswamy** [comment]
@yuruofeifei can you resolve conflicts and address all the comments, so that we can merge this PR.

**Comment by yuruofeifei** [comment]
@nswamy I've fixed the conflicts, Can you merge now

File: python/mxnet/base.py [hunk 5]
```diff
@@ -130,6 +142,15 @@ def c_array(ctype, values):
     -------
     out : ctypes array
         Created ctypes array.
+
+    Examples
+    --------
+    >>> x = mx.base.c_array(mx.base.mx_float, [1, 2, 3])
+    >>> print len(x)
+    3
+
```
Comment (piiswrong) [inline python/mxnet/base.py]: remove extra space.

File: python/mxnet/base.py (hunk 3 above)
Comment (piiswrong) [inline python/mxnet/base.py]: Pleas put mx_float into the following line.

**Comment by piiswrong** [comment]
Please address comment and we can merge

**Comment by yuruofeifei** [comment]
Fixed.

**Pull request merged**